torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -588,21 +614,23 @@ struct GraphFuser {
     }
 
     auto group = consumer;
-    if (consumer->kind() != prim::FusionGroup) {
+    if (consumer->kind() != kind_) {
       group = createSingletonFusionGroup(consumer);
     }
-    if (producer->node()->matches(
-            "aten::layer_norm(Tensor input, int[] normalized_shape, Tensor? weight, Tensor? bias, float eps, bool cudnn_enable) -> Tensor") ||
-        producer->node()->matches(
-            "aten::batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> Tensor")) {
+
+    if (kind_ == prim::FusionGroup &&
```

Comment:
This place is a bit meh, but I guess it's ok. It's a good sign that we might want to rethink the API you're adding because right now there's no way to do decompose ops lazily with custom fusions.